llm/server.go

```diff
@@ -0,0 +1,645 @@
+package llm
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"log/slog"
+	"math/rand"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/jmorganca/ollama/api"
+	"github.com/jmorganca/ollama/format"
+	"github.com/jmorganca/ollama/gpu"
+)
+
+// LlamaServer is an instance of the llama.cpp server
+type LlamaServer struct {
+	port int
+	cmd  *exec.Cmd
+	done chan error // Channel to signal when the process exits
+}
+
+func NewLlamaServer(model string, adapters, projectors []string, opts api.Options) (*LlamaServer, error) {
```

Comment:
A quick spot-check of this comparing against newDynExtServer looks like there's drift.  Do you think there's a plausible chance for regressions here and it's worth a deep comparison, or did you already refactor things to get this old code up to date?